docs/DesignDocs/ConfigIDInterface.md

```diff
@@ -0,0 +1,210 @@
+OpenEnclave Pre-measured Content Interface
+=======================================
+
+In this paper, we present the interface design for Pre-measured Content support in OpenEnclave. OpenEnclave Pre-measured Content refers to extra content that can be loaded into the enclave post enclave initialization, whose identity is recorded in enclave attestation evidence produced by the underlining enclave TEE environment.
+
+SGX CONFIGID and CONFIGSVN Overview
+-----------------
+
+Intel's 10th-gen Core processor and 3rd-gen Xeon-SP support the Key Separation
+and Sharing (KSS) feature, including `CONFIGID` and `CONFIGSVN`, which are new
+fields defined in `SECS`. `CONFIGID` and `CONFIGSVN` is intended to allow
+enclave creator to indicate what additional content may be accepted by the
+enclave post enclave initialization. The exact usage depends on the enclave
+implementation. `CONFIGSVN` might be used in case `CONFIGID` does not fully
+reflect the identity of the additional content. For example, `CONFIGID` can be
+set as the hash of the signing key or cert to verify the additional content, and
+`CONFIGSVN` can be set as the version number of the signed content. The
+`CONFIGID` and `CONFIGSVN` are part of the enclave identity produced by the CPU,
+reflecting the identity of the additional code/data allowed to be loaded into
+the enclave, committed at the enclave initialization time.
+
+Enclave Creation Interface
+-------------------------------------
+
+The Pre-measured Content concept is not limited to SGX TEE. Other TEEs can
+potentially support Pre-measured Content, as well as the design of using
+config_id/config_svn to indicate the identity of the extra content to be loaded
+into the enclave post enclave initialization.
+
+The enclave application passes config_id/config_svn to the enclave loader
+through the `OE_ENCLAVE_SETTING_CONTEXT_PRE_MEASURED_ID` type enclave setting
+context:
+
+```C
+/**
+ * Types of settings passed into **oe_create_enclave**
+ */
+typedef enum _oe_enclave_setting_type
+{
+    OE_ENCLAVE_SETTING_CONTEXT_SWITCHLESS = 0xdc73a628,
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+    OE_EXTENDED_ENCLAVE_INITIALIZATION_DATA = 0x976a8f66,
+#endif
+    /** Identity of additional content allowed to be loaded into the enclave
+     *  post enclave initialization. Currently only supported by SGX Enclaves
+     *  with KSS feature enabled.
+     */
+    OE_ENCLAVE_SETTING_CONTEXT_PRE_MEASURED_ID = 0x976a8f67,
+} oe_enclave_setting_type_t;
+
+typedef struct _oe_enclave_pre_measured_id
+{
+    /** The identity of additional content allowed to be loaded
+     *  into the enclave post enclave initialization. The identity is covered
+     *  by TEE generated Enclave Attestation Evidence.
+     */
+    uint8_t config_id[64];
+    uint16_t config_svn;
+}
+oe_enclave_pre_measured_id_t;
```

Comment:
What does "pre measured id" really mean?  Is the id itself before measurement?  Is the id an "identity" (like a key) or just an "identifier"?

Reply:
I will create a new PR for config_id/config_svn later, after the initial discussion already started in this PR reaches a good stopping point.

Reply:
Created [PR#3726](https://github.com/openenclave/openenclave/pull/3726) for config_id/config_svn.